### ISSUE TITLE
how to show chinese in Dolphin Smalltalk Professional?

### ISSUE THREAD

**Issue by ttkk1024** [dolphinsmalltalk/Dolphin#484]
When I input `Transcript show: '中文'` in Workspace,the code become `Transcript show: '??'`.how to slove the problem?Thank you very much.

**Comment by cDemersMitSci** [comment]
Dolphin Smalltak does not currently support Unicode in the UI.  It mostly uses ANSI APIs for UI and the String class is non-Unicode.  The solution would be to add Unicode support, and I have made a good start at that.

I have added Unicode support to most end user UI that Dolphin can present. I was not focused on adding Unicode support for development tools as I was mostly focused on an English language product that could be translated into Chinese.

I have a package that I plan to release that loads code (as normal) and also applies a series of programmatic transforms to sections of the base code to allow hooking of Unicode API calls in place of ANSI API calls.  I believe that I have most UI elements covered and also the ODBC API.  I think more work would be needed to add support for Unicode text in Smalltalk source code.

I only speak English and don't presently have an immediate need for a Chinese product translation.  So I am not in a great place to fully implement and test this.  I was hoping to get more people from the community involved.  I posted about it on the comp.lang.smalltalk.dolphin Newsgroup but didn't get any feedback.  I am not sure if there is much community interested in Unicode support for Dolphin anymore.  It used to come up from time to time in the past.

I want to make my Unicode UI package available on GitHub, but not sure where it best fits.  It isn't clean add-on in the sense that it makes significant system changes beyond the code it loads.  There is also some dependency work that I need to do.  When I have some free time I will give packaging it up another go.

I think Dolphin Next Generation (a commercial fork of Dolphin) added Unicode support but I don't know if it was ever released or ever will be.

**Comment by blairmcg** [comment]
Sounds good Chris. You should feel free to send PRs to replace the base system code if that makes it easier to ship your add on.

**Comment by cDemersMitSci** [comment]
I placed a PR to fix the "US Unicode" package to make it loadable in the Contributions repository.  Udo added enhanced capability to UnicodeString that my package needs.

No system changes are required to load my newest Unicode UI package, but it makes system changes once loaded.  I'll submit my package to the Contributions repository. I don't think it is in a position to replace base system code in this repository yet.  It needs more review from others.  I had to be a bit crafty in how I did some things to allow it to be loaded into unknown future versions of Dolphin and add Unicode support without being likely to break things.  It was an interesting puzzle.  A more formal persistent implementation may suggest different implementation choices.
